src/timestamp9.c

```diff
@@ -354,18 +354,19 @@ timestamp9_out(PG_FUNCTION_ARGS)
 	timestamp9 arg1 = PG_GETARG_TIMESTAMP9(0);
 	char *result = (char *) palloc(41);
 	time_t secs = (time_t)(arg1 / kT_ns_in_s);
-	struct tm tm_;
+	struct pg_tm *tm_;
 	size_t offset;
 	long long int mod = (arg1 % kT_ns_in_s);
 	if (mod < 0)
 	{
 		mod += kT_ns_in_s;
 		secs -= 1;
 	}
-	localtime_r(&secs, &tm_);
-	offset = strftime(result, 41, "%Y-%m-%d %H:%M:%S", &tm_);
+
+	tm_ = pg_localtime(&secs, session_timezone);
```

Comment:
Nice PR! There seems to be a possibility that pg_localtime returns NULL. Could you check for this and raise an error?

Reply:
Thanks for your kind reply and reviewing. I'm encountering regression tests failure since I'm in a different timezone :rofl: I think we can refactor some test cases with this patch landing. I can add some more tests for timestamp9.

Reply:
Thanks, it'd be great if you could refactor some of these tests to also work in the different timezones. The ability to use `set timezone` with this patch should actually help a lot with this.

Reply:
Thanks for your comments! I added more tests and tested them with Postgres REL_STABLE_12.

Reply:
Thanks for your contribution! I'll merge it.